### PR TITLE
bpo-44979: add pathlib.Path.relative (GH-27890)

### PR DESCRIPTION
**Pull request by cybergrind**
Enhancement to pathlib.Path that allows construction of path relative to caller's file

https://bugs.python.org/issue44979

### CODE REVIEW

File: Lib/pathlib.py [hunk 1]
```diff
@@ -1440,6 +1440,14 @@ def expanduser(self):
 
         return self
 
+    @classmethod
+    def relative(cls, path, depth=1):
```
Comment (ambv) [inline Lib/pathlib.py:1444]: 1. This name is too similar to `relative_to`.
2. The name doesn't explain it's relative to the caller.
3. It's not relative to the "caller file" but to the "caller directory".
4. The result isn't actually a relative path.

**Review by ambv — COMMENTED** [review]
I see what you're trying to do here but this looks like very specific functionality which doesn't belong in `pathlib` proper. I'm -1 to adding this.

I asked @pitrou for a review here as well since he's the author of PEP 428 and the biggest contributor to the library.

**Comment by cybergrind** [comment]
@ambv thank you for such fast feedback! 

> this looks like very specific functionality which doesn't belong in pathlib proper

Yeah, I'm not sure either if it should be here or not. It is more like paths work in js.

I can say that it is rather handy to have the ability to create a path relative to the current file:
1. when you're writing a script that can be invoked from an arbitrary directory but needs to read something near its location
2. when you have some fixtures in tests (json files, pictures and so on)

Right now it can be done in form `Path(__file__).parent / '../some_path'`, but we can provide better API for that.

File: Lib/pathlib.py (hunk 1 above)
Comment (cybergrind) [reply Lib/pathlib.py:1444]: Really cannot find a better name. It is `rel_path` in the internal package. 
I would really like to make it as short as possible, like `rel` but it isn't more descriptive at all.

**Comment by pitrou** [comment]
This seems like an extremely specific API. Perhaps you want to propose it for inclusion in [pytest](https://docs.pytest.org/en/), but I don't think it belongs in pathlib.

(also, it's not difficult to spell out `Path(__file__)`)

**Comment by pitrou** [comment]
I am rejecting this, sorry.

**Pull request closed without merging**